src/code.cloudfoundry.org/policy-server/asg_syncer/asg_syncer.go

```diff
@@ -0,0 +1,83 @@
+package asg_syncer
+
+//go:generate counterfeiter -generate
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/policy-server/cc_client"
+	"code.cloudfoundry.org/policy-server/store"
+	"code.cloudfoundry.org/policy-server/uaa_client"
+)
+
+//counterfeiter:generate -o fakes/asg_syncer.go --fake-name ASGSyncer . ASGSync
+type ASGSync interface {
+	Poll() error
+}
+
+type ASGSyncer struct {
+	Logger    lager.Logger
+	Store     store.SecurityGroupsStore
+	UAAClient uaa_client.UAAClient
+	CCClient  cc_client.CCClient
+}
+
+func NewASGSyncer(logger lager.Logger, store store.SecurityGroupsStore, uaaClient uaa_client.UAAClient, ccClient cc_client.CCClient) *ASGSyncer {
+	return &ASGSyncer{
+		Logger:    logger,
+		Store:     store,
+		UAAClient: uaaClient,
+		CCClient:  ccClient,
+	}
+}
+
+func (a *ASGSyncer) Poll() error {
+	a.Logger.Debug("asg-sync-started")
+	defer a.Logger.Debug("asg-sync-complete")
+
+	token, err := a.UAAClient.GetToken()
+	if err != nil {
+		return err
+	}
+	ccSGs, err := a.CCClient.GetSecurityGroups(token)
+	if err != nil {
+		return err
+	}
+
+	sgs := []store.SecurityGroup{}
+	for _, ccSG := range ccSGs {
+		stagingSpaces := []string{}
+		for _, data := range ccSG.Relationships.StagingSpaces.Data {
+			if guid, ok := data["guid"]; ok {
+				stagingSpaces = append(stagingSpaces, guid)
+			} else {
+				return fmt.Errorf("no 'guid' found for staging-space-relationship on asg '%s'", ccSG.GUID)
```

Comment:
This would get trigger if capi returns data in a new format, and we need to adjust. We want to error out and not update the definitions here. When moving to a syncer process, this should cause the process to fail